### PR TITLE
[Core][Python3.10] Refactor codebase for Python 3.10 compatibility, removing Python 2.7 related stuff

### PR DESCRIPTION
**Pull request by agl29**
## What changes were proposed in this pull request?

- Support for python3.10
- Removed python2.7 related code from makefiles

## How was this patch tested?

- Locally
- Able to build Hue locally with python3.8, python3.9 and python3.10
- CI check

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -133,30 +131,14 @@ endif
...
-	@echo "--- start installing PIP_MODULES in virtual-env"
-	@$(ENV_PIP) install --upgrade pip
-	@$(ENV_PIP) install --upgrade --force-reinstall $(PIP_MODULES)
-	@echo "--- done installing PIP_MODULES in virtual-env"
-endif
-
-.PHONY: virtual-env3.8
-virtual-env3.8: $(BLD_DIR_ENV)/stamp
+.PHONY: virtual-env
+virtual-env: $(BLD_DIR_ENV)/stamp
```
Comment (Harshg999) [inline Makefile:137]: Do we require line 133, 134 also with this present?

**Review by Harshg999 — APPROVED** [review]
LGTM!

**Comment by wing2fly** [comment]
@agl29 
Can you please make .github/workflows/commitflow-py3.yml as separate commits? This is upstream only needed.

**Review by wing2fly — APPROVED** [review]

**Review by ranade1 — COMMENTED** [review]
Have you checked on PPC Python 3.8 and PPC Python 3.9 system? How does it work on there?

**Review by ranade1 — APPROVED** [review]
Thanks for incorporating my review suggestions. Hope you have tested this change on compatible Python versions?

**Comment by agl29** [comment]
> Thanks for incorporating my review suggestions. Hope you have tested this change on compatible Python versions?

Yes, tested with python 3.8, 3.9 and 3.10

**Pull request merged**